lib/OpenQA/Scheduler/Model/Jobs.pm

```diff
@@ -43,8 +43,8 @@ sub schedule ($self, $allocated_workers = {}, $allocated_jobs = {}) {
     my $worker_count = $schema->resultset('Workers')->count;
     my $free_worker_count = @$free_workers;
     my $running = $schema->resultset('Jobs')->count({state => [OpenQA::Jobs::Constants::EXECUTION_STATES]});
-    my $limit = $self->{config}->{scheduler}->{max_running_jobs} // '';
-    if (length $limit and $running >= $limit) {
+    my $limit = $self->{config}->{scheduler}->{max_running_jobs};
+    if ($limit >= 0 and $running >= $limit) {
```

Comment:
I really like `and` because of it's clear lower precedence, but you won't give up :)

Reply:
> I'd also avoid a "x / y" format unless we are really sure not to end up with e.g. "201 / 200" in production (which would look rather buggy).

Didn't we discuss that the code path is already behind checks for the number of running jobs? It seems odd to me to avoid it just because we may not know about bugs that break it.

Reply:
We could very well end up with 201 / 200, for example when we introduce the limit or lower it and the number of running jobs is still higher.

Btw., the comment seems misplaced here. This review comment from @Martchus is about `and` vs. `&&`.

Reply:
You pointed me to this comment ;-)

Anyway feel free to override me if you're on the same page with Marius

Reply:
No, I pointed to https://github.com/os-autoinst/openQA/pull/5279#issuecomment-1673322996

Reply:
To clarify: It would show `250 / 250 jobs running` and possibly sometimes `260 / 250`, but never `200 / 250`, so Marius and I really thing that wouldn't make sense.
Meanwhile in the meeting Marius and I also decided that it's best to only show the notice if the query is not limited by `match` or `group_id` parameters. Because then it would show e.g. `3 jobs running (limited by server config)`, even if there are 250 jobs in total.
Will push an update later today.